app/src/main/java/com/example/dispatcher/domain/homepage/repo/ArticleRepository.kt

```diff
@@ -0,0 +1,30 @@
+package com.example.dispatcher.domain.homepage.repo
+
+import android.content.Context
+import com.example.dispatcher.R
+import com.example.dispatcher.presentation.homepage.model.Article
+import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+import java.io.InputStreamReader
+
+class ArticleRepository(context: Context) {
```

Comment:
Create an interface for the repo and also a factory to control and manage the implementation of this interface.